sql/core/src/test/scala/org/apache/spark/sql/execution/metric/SQLMetricsSuite.scala

```diff
@@ -194,10 +194,26 @@ class SQLMetricsSuite extends SparkFunSuite with SQLMetricsTestUtils with Shared
   }
 
   test("Sort metrics") {
-    // Assume the execution plan is
-    // WholeStageCodegen(nodeId = 0, Range(nodeId = 2) -> Sort(nodeId = 1))
-    val ds = spark.range(10).sort('id)
-    testSparkPlanMetrics(ds.toDF(), 2, Map.empty)
+    // Assume the execution plan with node id is
+    // Sort(nodeId = 0)
+    //   Exchange(nodeId = 1)
+    //     Project(nodeId = 2)
+    //       LocalTableScan(nodeId = 3)
+    // Because of SPARK-25267, ConvertToLocalRelation is disabled in the test cases of sql/core,
+    // so Project here is not collapsed into LocalTableScan.
+    val df = Seq(1, 3, 2).toDF("id").sort('id)
+    val metrics = getSparkPlanMetrics(df, 2, Set(0))
+    val sortMetrics = metrics.get.get(0).get
+    // Check node 0 is Sort node
+    val operatorName = sortMetrics._1
+    assert(operatorName == "Sort")
+    // Check metrics values
+    val sortTimeStr = sortMetrics._2.get("sort time total (min, med, max)").get.toString
+    timingMetricStats(sortTimeStr).foreach { case (sortTime, _) => assert(sortTime >= 0) }
```

Comment:
According to SortExec, sort time may be 0 because it is converted from nano-seconds to milli-seconds.
https://github.com/apache/spark/blob/4ee2c8d7de4024dae9605e4ad18a8f541a234625/sql/core/src/main/scala/org/apache/spark/sql/execution/SortExec.scala#L109

I tried to run with "sortTime > 0", sometimes it failed.
<img width="780" alt="screen shot 2018-12-26 at 10 27 00 am" src="https://user-images.githubusercontent.com/12194089/50429497-c6669000-08f9-11e9-904b-08250e6fe71d.png">

Reply:
nit: `assert(timingMetricStats(sortTimeStr).forall { case (sortTime, _) => sortTime >= 0 })`?

Reply:
Yes, it's better.

Reply:
Fixed with a new commit.

Reply:
I see, can we add a comment with the reason why it can be 0?